main/src/test/java/org/mobilitydata/gtfsvalidator/validator/CalendarServiceDateValidatorTest.java

```diff
@@ -0,0 +1,176 @@
+        verify(mockCalendar, times(1)).startDate();
+        verify(mockStartDate, times(1)).isAfter(mockEndDate);
+        verifyNoMoreInteractions(mockCalendar, mockEndDate, mockStartDate);
+    }
+
+    @Test
+    public void startDateAfterEndDateShouldGenerateNotice() {
+        NoticeContainer mockNoticeContainer = mock(NoticeContainer.class);
+        List<GtfsCalendar> calendarCollection = new ArrayList<>();
+        GtfsCalendar mockCalendar = mock(GtfsCalendar.class);
+        when(mockCalendar.hasEndDate()).thenReturn(true);
+        when(mockCalendar.hasStartDate()).thenReturn(true);
+
+        GtfsDate mockStartDate = mock(GtfsDate.class);
+        GtfsDate mockEndDate = mock(GtfsDate.class);
+
+        when(mockStartDate.isAfter(mockEndDate)).thenReturn(true);
+        when(mockCalendar.startDate()).thenReturn(mockStartDate);
+        when(mockStartDate.toYYYYMMDD()).thenReturn("start date value");
+        when(mockEndDate.toYYYYMMDD()).thenReturn("end date value");
+        when(mockCalendar.endDate()).thenReturn(mockEndDate);
+        when(mockCalendar.csvRowNumber()).thenReturn(2L);
+        when(mockCalendar.serviceId()).thenReturn("service id value");
+        calendarCollection.add(mockCalendar);
+        when(mockCalendarTable.getEntities()).thenReturn(calendarCollection);
+        when(mockCalendarTable.gtfsFilename()).thenReturn("calendar.txt");
+
+        underTest.validate(mockNoticeContainer);
+
+        ArgumentCaptor<StartAndEndDateOutOfOrderNotice> captor =
+                ArgumentCaptor.forClass(StartAndEndDateOutOfOrderNotice.class);
+
+        verify(mockNoticeContainer, times(1)).addNotice(captor.capture());
+        StartAndEndDateOutOfOrderNotice notice = captor.getValue();
+
+        assertThat(notice.getCode()).matches("start_and_end_date_out_of_order");
+        assertThat(notice.getContext()).containsEntry("filename", "calendar.txt");
+        assertThat(notice.getContext()).containsEntry("csvRowNumber", 2L);
+        assertThat(notice.getContext()).containsEntry("entityId", "service id value");
+        assertThat(notice.getContext()).containsEntry("startDate", "start date value");
+        assertThat(notice.getContext()).containsEntry("endDate", "end date value");
+
+        verify(mockCalendar, times(1)).hasEndDate();
+        verify(mockCalendar, times(1)).hasStartDate();
+        verify(mockCalendar, times(1)).serviceId();
+        verify(mockCalendarTable, times(1)).gtfsFilename();
+        verify(mockCalendar, times(1)).csvRowNumber();
+        verify(mockCalendar, times(2)).endDate();
+        verify(mockCalendar, times(2)).startDate();
+        verify(mockStartDate, times(1)).isAfter(mockEndDate);
+        verify(mockStartDate, times(1)).toYYYYMMDD();
+        verify(mockEndDate, times(1)).toYYYYMMDD();
+        verifyNoMoreInteractions(mockCalendar, mockEndDate, mockStartDate);
+    }
+
+    @Test
+    public void noStartDateShouldNotGenerateNotice() {
```

Comment:
Changing this name would require us to change the validator implementation since it actually does not throw exception when `calendar.start_date` is null. So because of that I am not sure that is out best option.

However, yes - adding comment in the validator implementation would improve the code clarity.

Reply:
>Changing this name would require us to change the validator implementation since it actually does not throw exception when calendar.start_date is null..so I am not sure that is out best option.

Or maybe `...ShouldRunWithoutError()`? It's still testing that the case where the field is missing doesn't crash the validator.